core/src/main/java/org/neo4j/graphalgo/core/utils/paged/HugeArrays.java

```diff
@@ -2,7 +2,7 @@
 
 final class HugeArrays {
 
-    private static final int SINGLE_PAGE_SHIFT = 30;
+    private static final int SINGLE_PAGE_SHIFT = 28;
```

Comment:
why this change? was this a bug?

Reply:
allocating a large array might trigger a gc cycle, possibly full, to make room for the array (long, doubles, objects require 8 consecutive gigs with a shift of 30) and we have no way of figuring out if this is going to happen – It might also fail even when there seems to be enough heap available and it takes a while to allocate an array that long. With 28, we just bail out earlier to the paged version to not trigger those edge cases that often.